src/compiler/checker.ts

```diff
@@ -21455,6 +21461,10 @@ namespace ts {
         }
 
         function errorNextVariableOrPropertyDeclarationMustHaveSameType(firstDeclaration: Declaration, firstType: Type, nextDeclaration: Declaration, nextType: Type): void {
+            if (isIdentifier(firstDeclaration) || isIdentifier(nextDeclaration)) {

```

Comment:
Again, I really feel like this should be a flag on the symbol and not be information implicitly derived from the declaration being an identifier.